### PR TITLE
Implement `cml runner launch --cloud-image`

### PR DESCRIPTION
**Pull request by 0x2b3bfa0**
Closes https://github.com/iterative/cml/issues/1285 and discord#cml/1057160051904753704 (wishful thinking, needs testing)

* Amazon Web Services: AMI name, like `iterative-cml` (potentially dangerous, allows overriding by a third party)
* Google Cloud: machine image families in the `project/family` format, like `ubuntu-os-cloud/ubuntu-2004-lts`
* Microsoft Azure: machine images in the `publisher:offer:sku:version` format, like `Canonical:UbuntuServer:18.04-LTS:latest`
* Kubernetes: valid [container image names](https://kubernetes.io/docs/concepts/containers/images/#image-names) like `ubuntu` or `ghcr.io/user/project:tag`

### CODE REVIEW

**Review by DavidGOrtega — APPROVED** [review]
lgtm

Here this is easy, however the user should now that the cml stack relies heavily on Debian? A better option name might be `--cloud-image-based-on-debian` 😁 ?

**Comment by dacbd** [comment]
a cml-playground minimal example that we can at least link to?

**Comment by 0x2b3bfa0** [comment]
@DavidGOrtega, that’s better solved with documentation… or lack thereof. 😈 

https://github.com/iterative/cml/blob/3bf7095448449b7f5753a3833824164ac7c7d608/bin/cml/runner/launch.js#L590

**Comment by dacbd** [comment]
retroactively tested / gcp/aws examples: https://github.com/iterative/cml-playground/pull/312

File: bin/cml/runner/launch.js [hunk 1]
```diff
@@ -582,6 +584,11 @@ exports.options = kebabcaseKeys({
     description: 'Specifies the subnet to use within AWS',
     alias: 'cloud-aws-subnet-id'
   },
+  cloudImage: {
+    type: 'string',
+    description: 'Custom machine/container image',
+    hidden: true
```
Comment (casperdcl) [inline bin/cml/runner/launch.js:590]: qq @0x2b3bfa0  - should this be documented in https://cml.dev/doc/ref/runner?

Comment (0x2b3bfa0) [reply bin/cml/runner/launch.js:590]: Because https://github.com/iterative/cml/pull/1302#issuecomment-1369117730

Comment (0x2b3bfa0) [reply bin/cml/runner/launch.js:590]: Scared to make that part of the public API and having to maintain a footgun.

Comment (casperdcl) [reply bin/cml/runner/launch.js:590]: I can understand:

1. implementing X.
2. not implementing X.
3. implementing X but hiding it as a beta feature.

Sounds like `--cloud-image` falls into none of the above categories? I'm confused.

Comment (0x2b3bfa0) [reply bin/cml/runner/launch.js:590]: Sounds like 3 to me: implemented but not officially supported for now,

**Pull request merged**